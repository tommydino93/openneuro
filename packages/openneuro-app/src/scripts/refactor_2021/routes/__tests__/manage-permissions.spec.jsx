import React from 'react'
import { shallow } from 'enzyme'
import { PermissionRow, ShareTable } from '../manage-permissions.jsx'

describe('Manage dataset permissions route', () => {
  describe('PermissionRow', () => {
    it('renders for admin permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="admin"
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders for ro permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="ro"
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders for rw permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="rw"
          />,
        ),
      ).toMatchSnapshot()
    })
  })
  describe('ShareTable', () => {
    const id = 'ds001'
    it('renders with no permissions', () => {
      expect(
        shallow(
          <ShareTable
            datasetId={id}
            permissions={{ id, userPermissions: [] }}
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders with one permission', () => {
      expect(
        shallow(
          <ShareTable
            datasetId={id}
            permissions={{
              id,
              userPermissions: [
                {
                  user: { id: '1234', email: 'test@example.com' },
                  access: 'ro',
                },
              ],
            }}
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders with many permissions', () => {
      expect(
        shallow(
          <ShareTable
            datasetId={id}
            permissions={{
              id,
              userPermissions: [
                {
                  user: { id: '1234', email: 'test@example.com' },
                  access: 'ro',
                },
                {
                  user: { id: '5678', email: 'tester@example.com' },
                  access: 'rw',
                },
                {
                  user: { id: '91011', email: 'testr@example.com' },
                  access: 'admin',
                },
              ],
            }}
          />,
        ),
      ).toMatchSnapshot()
    })
  })
})
