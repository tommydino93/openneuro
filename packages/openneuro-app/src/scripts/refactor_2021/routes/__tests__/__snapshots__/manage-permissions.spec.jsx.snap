// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage dataset permissions route PermissionRow renders for admin permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    Edit dataset and edit permissions
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;

exports[`Manage dataset permissions route PermissionRow renders for ro permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    View dataset
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;

exports[`Manage dataset permissions route PermissionRow renders for rw permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    Edit dataset
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;

exports[`Manage dataset permissions route ShareTable renders with many permissions 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
      <th>
        Access
      </th>
      <th>
        Edit
      </th>
    </tr>
  </thead>
  <tbody>
    <PermissionRow
      datasetId="ds001"
      key="0"
      userEmail="test@example.com"
      userId="1234"
    />
    <PermissionRow
      datasetId="ds001"
      key="1"
      userEmail="tester@example.com"
      userId="5678"
    />
    <PermissionRow
      datasetId="ds001"
      key="2"
      userEmail="testr@example.com"
      userId="91011"
    />
  </tbody>
</table>
`;

exports[`Manage dataset permissions route ShareTable renders with no permissions 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
      <th>
        Access
      </th>
      <th>
        Edit
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Manage dataset permissions route ShareTable renders with one permission 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
      <th>
        Access
      </th>
      <th>
        Edit
      </th>
    </tr>
  </thead>
  <tbody>
    <PermissionRow
      datasetId="ds001"
      key="0"
      userEmail="test@example.com"
      userId="1234"
    />
  </tbody>
</table>
`;
